provisioning/image_cache.go

```diff
@@ -137,13 +137,14 @@ func newImageCacheContainers(images *Images, proxy *osconfigv1.Proxy) []corev1.C
 	return injectProxyAndCA(containers, proxy)
 }
 
-func newImageCachePodTemplateSpec(targetNamespace string, images *Images, provisioningConfig *metal3iov1alpha1.ProvisioningSpec, proxy *osconfigv1.Proxy) (*corev1.PodTemplateSpec, error) {
-	cacheConfig, err := imageCacheConfig(targetNamespace, *provisioningConfig)
+func newImageCachePodTemplateSpec(info *ProvisioningInfo) (*corev1.PodTemplateSpec, error) {
+	cacheConfig, err := imageCacheConfig(info.Namespace, info.ProvConfig.Spec)
 	if err != nil {
 		return nil, err
 	}
+	info.ProvConfig.Spec = *cacheConfig
```

Comment:
This is very dangerous, because it sticks the locally-modified Spec that is used just for the image cache into the global ProvisioningInfo that is passed to every function. We're only getting away with it because setting up the image cache is currently the last function called.

Reply:
https://github.com/openshift/cluster-baremetal-operator/pull/179